endpoints/openrtb2/auction_test.go

```diff
@@ -897,6 +897,28 @@ func TestRefererParsing(t *testing.T) {
 	}
 }
 
+func TestSitePageParsing(t *testing.T) {
+	httpReq := httptest.NewRequest("POST", "/openrtb2/auction", strings.NewReader(validRequest(t, "site.json")))
+	bidReq := &openrtb2.BidRequest{
+		Site: &openrtb2.Site{
+			Page: "https://test.somepage.com",
+		},
+	}
+
```

Comment:
Hi @VeronikaSolovei9,

Thank you a lot for he review. I've added two new testcases as well as the comments for it:
- http.referer and site.page are set, site.domain is not set;
- http.referer, site.page and site.domain are set.

Reply:
Hi @vfedoseev,

Thanks for adding the test cases. The coverage looks good. It looks like you've copied the style of the older referer test case `TestRefererParsing` just above these. Given that this test was written at prebid server's inception and uses an antiquated assertion style before the introduction of the `testify` package, would you mind consolidating your tests and `TestRefererParsing` into a table-based test that swaps out the `t.Fatalf` and `t.Errorf` assertions for `assert.NotNil` and `assert.Equal`?

Something like this would work:
```
func TestReferer(t *testing.T) {
	tests := []struct{
		description     string
		givenSitePage   string
		givenSiteDomain string
		givenReferer    string
		expectedDomain  string
		expectedPage    string
	}{
		{
			description:    "site.page/domain are unchanged when site.page/domain and http referer are not set",
			expectedDomain: "",
			expectedPage:   "",
		},
		{
			description:    "site.page/domain are derived from referer when neither is set and http referer is set",
			givenReferer:   "https://test.somepage.com",
			expectedDomain: "somepage.com",
			expectedPage:   "https://test.somepage.com",
		},
		{
			description:    "site.domain is derived from site.page when site.page is set and http referer is not set",
			givenSitePage:  "https://test.somepage.com",
			expectedDomain: "somepage.com",
			expectedPage:   "https://test.somepage.com",
		},
		{
			description: "site.domain is derived from http referer when site.page and http referer are set",
			givenSitePage: "https://test.somepage.com",
			givenReferer:  "http://test.com",
			expectedDomain: "test.com",
			expectedPage: "https://test.somepage.com",
		},
		{
			description: "site.page/domain are unchanged when site.page/domain and http referer are set",
			givenSitePage: "https://test.somepage.com",
			givenSiteDomain: "some.domain.com",
			givenReferer:  "http://test.com",
			expectedDomain: "some.domain.com",
			expectedPage: "https://test.somepage.com",
		},
	}

	for _, tt := range tests {
		httpReq := httptest.NewRequest("POST", "/openrtb2/auction", strings.NewReader(validRequest(t, "site.json")))
		httpReq.Header.Set("Referer", tt.givenReferer)

		bidReq := &openrtb2.BidRequest{
			Site: &openrtb2.Site{
				Domain: tt.givenSiteDomain,
				Page: tt.givenSitePage,
			},
		}

		setSiteImplicitly(httpReq, bidReq)

		assert.NotNil(t, bidReq.Site, tt.description)
		assert.Equal(t, tt.expectedDomain, bidReq.Site.Domain, tt.description)
		assert.Equal(t, tt.expectedPage, bidReq.Site.Page, tt.description)
	}
}
```

Reply:
Hi @bsardo,

Thank you a lot for the review and for the suggestions.
I've combined all the tests related to the domain/referer parsing to table-based test.

Looking forward to your reply.